pkg/operation/botanist/component/coredns/coredns.go

```diff
@@ -528,6 +536,138 @@ import custom/*.server
 				},
 			},
 		}
+
+		clusterProportionalDNSAutoscalerServiceAccount = &corev1.ServiceAccount{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "coredns-autoscaler",
+				Namespace: metav1.NamespaceSystem,
+			},
+			AutomountServiceAccountToken: pointer.Bool(false),
+		}
+
+		clusterProportionalDNSAutoscalerClusterRole = &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "system:coredns-autoscaler",
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups: []string{""},
+					Resources: []string{"nodes"},
+					Verbs:     []string{"list", "watch"},
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"replicationcontrollers/scale"},
+					Verbs:     []string{"get", "update"},
+				},
+				{
+					APIGroups: []string{"apps"},
+					Resources: []string{"deployments/scale", "replicasets/scale"},
+					Verbs:     []string{"get", "update"},
+				},
+				// Remove the configmaps rule once below issue is fixed:
+				// kubernetes-incubator/cluster-proportional-autoscaler#16
+				{
+					APIGroups: []string{""},
+					Resources: []string{"configmaps"},
+					Verbs:     []string{"get", "create"},
+				},
+			},
+		}
+
+		clusterProportionalDNSAutoscalerClusterRoleBinding = &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "system:coredns-autoscaler",
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterProportionalDNSAutoscalerClusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      clusterProportionalDNSAutoscalerServiceAccount.Name,
+				Namespace: clusterProportionalDNSAutoscalerServiceAccount.Namespace,
+			}},
+		}
+
+		clusterProportionalDNSAutoscalerDeployment = &appsv1.Deployment{
```

Comment:
Can you add a `VPA` for this deployment in case the shoot enables VPA?

Reply:
Added vertical pod autoscaler in latest commit.
However, it feels somewhat strange to autoscale the autoscaler. It also makes the values more confusing.